### PR TITLE
Updated node deps for package.json

### LINKED ISSUE #1743: Update node deps
We should look at all our dependencies in `package.json` and update them.

See also https://github.com/mozilla/brackets/issues/589

### PR DESCRIPTION
**Pull request by omytryniuk**
fixes #1743

I have updated the dependencies. 
I have tested "npm install":

- without virtualization
- running Thimble on VM as well. No errors. 

Thank you for your advice to use npm-check, it was a very useful tool in terms of checking updated versions of dependencies. 

I have not had any problems, except grunt, and that's why I decided to stick with same version of the dependency we had before (however most of grunt peerdependencies were updated).

Although I did not see any errors, I would really like you, if you have a chance, to run it as well, to be sure that there are no errors. 

Thank you!
screenshot 2017-03-07 17 55 13

### CODE REVIEW

**Comment by omytryniuk** [comment]
@Pomax, @gideonthomas I spent some time trying to figure out why Travis CI does not pass the test. 
First of all, I was using the newest nodejs version(6.9.5) on my machine, comparing to the one Travis CI runs on github(4.6.2). When I tested updated package.json on my machine, I did not have any issues. 
However, now I am using the same version as on github(4.6.2), and I was able to reproduce the errors Travis CI shows. 
The problem was with this package: "grunt-contrib-jshint": "^1.1.0". 
According to the error description, the newest version of jshint treats the next code as an error:

```js
var habitat = require('habitat');
habitat.load(require('path').resolve(__dirname, '../../.env'));
var env = new habitat();
```

Because constructor is supposed to start with an upper case.

I have updated dependencies, however I could not pass Travis test on github. As I guess, it is because of the cache created by Travis CI when I was runnning Travis CI the first time. 
I think, it is because node_modules still contains "grunt-contrib-jshint": "^1.1.0". Even I downgraded the version to ^0.11.2 in package.json, it still keeps the highest one. 

I see cleaning cache (node_modules) and installing packages afterwards as a possible solution.
I am really sorry about that. 

I would also like to ask. I have updated major versions for some of the dependencies and do not see any errors. However, if you think that it is better practice to update only minor versions, I can do that. 

Thank you again for you help and sorry for the inconvenience.

P.S. I attach the screenshot that shows there is no error when I run on VM (again, node_modules where deleted before the installation)
proof

**Comment by Pomax** [comment]
Looks like there are some linting errors with your latest commit, can you run `npm test` locally and fix any issues that flags?

**Comment by humphd** [comment]
He's got it fixed in https://github.com/mozilla/thimble.mozilla.org/pull/1847

**Comment by Pomax** [comment]
Ah, will we be landing sequencially with a rebase?

**Comment by humphd** [comment]
Now that #1847 has landed, let's rebase this and see if we can proceed.

File: package.json [hunk 1]
```diff
@@ -24,56 +24,56 @@
...
+    "compression": "^1.6.2",
+    "cookie-parser": "^1.4.3",
     "cookie-session": "^1.1.0",
     "cookies-js": "~1.2.1",
-    "cors": "^2.7.1",
-    "cryptr": "^1.0.0",
-    "csurf": "^1.8.3",
-    "express": "^4.13.0",
+    "cors": "^2.8.1",
+    "cryptr": "^2.0.0",
```
Comment (Pomax) [inline package.json]: are we sure this is safe? minor version bumps (assuming correct `semver` use) are supposedly safe but major version bumps are API-compatibility-breaking changes.

Let's keep this at `^1.0.0` for now and file a separate issue for cryptr specifically. Going up major version numbers is worth "doing on its own"

File: package.json [hunk 2]
```diff
@@ -24,56 +24,56 @@
...
     "grunt-contrib-jshint": "^0.11.2",
-    "grunt-contrib-less": "^1.0.1",
-    "grunt-contrib-requirejs": "^0.4.4",
-    "grunt-execute": "0.1.5",
+    "grunt-contrib-less": "^1.4.0",
+    "grunt-contrib-requirejs": "^1.0.0",
+    "grunt-execute": "0.2.2",
     "grunt-git": "^0.3.4",
-    "grunt-lesslint": "^1.1.14",
+    "grunt-lesslint": "^3.2.0",
```
Comment (Pomax) [inline package.json]: can we keep this at `^1.1.14` and file a separate issue for grunt-lesslint specifically? Going up by two major version numbers is worth "doing on its own"

File: package.json [hunk 3]
```diff
@@ -24,56 +24,56 @@
...
     "grunt-npm": "git://github.com/sedge/grunt-npm.git#branchcheck",
     "grunt-update-submodules": "^0.4.1",
     "habitat": "3.1.2",
-    "helmet": "2.1.2",
-    "jit-grunt": "^0.9.1",
-    "jquery": "~2.1.4",
-    "less-middleware": "2.0.1",
+    "helmet": "3.4.1",
```
Comment (Pomax) [inline package.json]: Let's keep this at `2.1.2` for now and file a separate issue for helmet specifically. Going up a major version number for a security-related library is worth "doing on its own"

File: package.json [hunk 4]
```diff
@@ -24,56 +24,56 @@
...
     "throng": "^4.0.0",
-    "time-grunt": "^1.2.1",
-    "uuid": "^2.0.1",
-    "webmaker-analytics": "~0.1.5",
+    "time-grunt": "^1.4.0",
+    "uuid": "2.0.3",
+    "webmaker-analytics": "~0.1.7",
     "webmaker-i18n": "^0.3.31",
-    "wolfy87-eventemitter": "~4.3.0"
+    "wolfy87-eventemitter": "~5.1.0"
```
Comment (Pomax) [inline package.json]: I have no idea what this is used for, which makes it all the more important that we don't bump this up a major version number outright - let's file a separate issue for this dependency, too.

**Comment by humphd** [comment]
@omytryniuk can you make the changes @Pomax requested so we can move this forward?

**Comment by omytryniuk** [comment]
@humphd, @Pomax  My apologies for keeping you waiting so long. There were some network issues. I have downgraded the versions you have asked for and I also think it is a good idea to file separate issues for some dependencies.

**Comment by Pomax** [comment]
gah! equal apologies for taking so long to review - it looks like we forgot to flag a few more major versions

File: package.json [hunk 5]
```diff
@@ -24,54 +24,54 @@
...
+    "cors": "^2.8.1",
     "cryptr": "^1.0.0",
-    "csurf": "^1.8.3",
-    "express": "^4.13.0",
+    "csurf": "^1.9.0",
+    "express": "^4.15.2",
     "form-data": "^0.2.0",
-    "fs-writefile-promise": "1.0.3",
-    "glob": "^6.0.4",
+    "fs-writefile-promise": "2.0.0",
```
Comment (Pomax) [inline package.json]: let's undo this major version

File: package.json [hunk 6]
```diff
@@ -24,54 +24,54 @@
...
+    "glob": "^7.1.1",
```
Comment (Pomax) [inline package.json]: let's undo this major version

File: package.json [hunk 7]
```diff
@@ -24,54 +24,54 @@
...
     "grunt": "~0.4.1",
-    "grunt-cli": "^0.1.13",
-    "grunt-contrib-csslint": "0.1.2",
+    "grunt-cli": "^1.2.0",
```
Comment (Pomax) [inline package.json]: let's undo this major version

File: package.json [hunk 8]
```diff
@@ -24,54 +24,54 @@
...
+    "grunt-contrib-csslint": "2.0.0",
```
Comment (Pomax) [inline package.json]: let's undo this major version

File: package.json [hunk 9]
```diff
@@ -24,54 +24,54 @@
...
     "grunt-lesslint": "^1.1.14",
     "grunt-npm": "git://github.com/sedge/grunt-npm.git#branchcheck",
     "grunt-update-submodules": "^0.4.1",
     "habitat": "3.1.2",
     "helmet": "2.1.2",
-    "jit-grunt": "^0.9.1",
-    "jquery": "~2.1.4",
-    "less-middleware": "2.0.1",
+    "jit-grunt": "^0.10.0",
+    "jquery": "~3.1.1",
```
Comment (Pomax) [inline package.json]: let's undo this major version

**Comment by Pomax** [comment]
I've filed https://github.com/mozilla/thimble.mozilla.org/issues/1908 to track the major version dependencies.

**Comment by omytryniuk** [comment]
@Pomax, no worries:). Thank you very much for the comments! I have downgraded as you have asked. Thank you for creating the issue for the major version dependencies.

**Comment by Pomax** [comment]
With that, this looks good to land to me.

**Comment by omytryniuk** [comment]
@Pomax, Thank you very much for your help!

**Pull request merged**